discovery_engine_core/core/src/engine.rs

```diff
@@ -158,12 +161,15 @@ where
             .map_err(Error::InvalidStack)?;
         let core_config = CoreConfig::default();
 
-        Ok(Self {
+        let mut engine = Self {
             config,
             core_config,
             stacks,
             ranker,
-        })
+        };
+        engine.update_stacks(0).await?;
```

Comment:
This will update the stacks only if they are empty but we want to update them whatever is their size.

Reply:
you are absolutely right, i'll change it to `usize::MAX`, i shouldn't work that late oO